src/main/java/io/jenkins/plugins/gitlabbranchsource/GitLabMergeRequestCommentTrigger.java

```diff
@@ -56,29 +58,33 @@ public void isMatch() {
                         if (gitLabSCMSource.getProjectId() == getPayload().getMergeRequest().getTargetProjectId()
                                 && isTrustedMember(gitLabSCMSource, sourceContext.getOnlyTrustedMembersCanTrigger())) {
                             for (Job<?, ?> job : owner.getAllJobs()) {
```

Comment:
With this for, we loop to all jobs. Even out of the projectId. And there isn't more check after with the payload